java/src/jmri/jmrix/openlcb/swing/hub/HubAction.java

```diff
@@ -10,14 +10,29 @@ public HubAction() {
         super("LCC Hub Control",
                 new JmriJFrameInterface(),
                 HubPane.class.getName(),
-                jmri.InstanceManager.getNullableDefault(CanSystemConnectionMemo.class));
+                memo);
     }
 
     public HubAction(String name, WindowInterface  iface) {
         super(name,
                 iface,
                 HubPane.class.getName(),
-                jmri.InstanceManager.getNullableDefault(CanSystemConnectionMemo.class));
+                memo);
     }
 
+    // we have to locate the (only) LCC or OpenLCB CanSystemConnectionMemo
+    // before we can call the super constructors.
+    private static CanSystemConnectionMemo memo;
+    
+    static {        
+        var memos = jmri.InstanceManager.getList(CanSystemConnectionMemo.class);
+        for (CanSystemConnectionMemo check : memos) {
+            String name = check.getUserName();
+            if (name.equals("LCC") || name.equals("OpenLCB")) {
```

Comment:
Will this work? If I understand the code correctly, the username can be set by the user.